src/main/java/core/basesyntax/service/FileWriterImpl.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.service;
+
+import java.io.IOException;
+
+public class FileWriterImpl implements FileWriter {
+    @Override
+    public void writeData(String report, String fileName) {
+        try (java.io.FileWriter writer = new java.io.FileWriter(fileName)) {
+            writer.write(report);
+        } catch (IOException e) {
+            System.err.println("Помилка запису у файл: " + e.getMessage());
```

Comment:
Instead of printing the error message to `System.err`, consider throwing a custom exception or logging the error using a logging framework. This will provide better error handling and make it easier to debug issues in a production environment.